consensus/src/alerts/service.rs

```diff
@@ -157,26 +197,70 @@ impl<H: Hasher, D: Data, MK: MultiKeychain> Service<H, D, MK> {
         handler: &mut Handler<H, D, MK>,
         multisigned: Multisigned<H::Hash, MK>,
     ) {
-        match handler.alert_confirmed(multisigned) {
-            Ok(notification) => self.send_notification_for_units(notification),
+        match handler.alert_confirmed(multisigned.clone()) {
+            Ok(notification) => {
+                self.multisigned_notifications
+                    .insert(*multisigned.as_signable(), notification);
+                if self
+                    .data_for_backup
+                    .unbounded_send(AlertData::MultisignedHash(multisigned))
+                    .is_err()
+                {
+                    error!(
+                        target: LOG_TARGET,
+                        "{:?} multisigned hash couldn't be sent to backup.", self.node_index
+                    );
+                }
+            }
             Err(error) => warn!(target: LOG_TARGET, "{}", error),
         }
     }
 
+    fn handle_data_from_backup(&mut self, data: AlertData<H, D, MK>) {
+        match data {
+            AlertData::OwnAlert(alert) => {
+                if let Some((message, recipient, hash)) =
+                    self.own_alert_responses.remove(&alert.hash())
+                {
+                    self.send_message_for_network(message, recipient);
+                    self.rmc.start_rmc(hash);
+                }
+            }
+            AlertData::NetworkAlert(alert) => {
+                if let Some((maybe_notification, hash)) =
+                    self.network_alert_responses.remove(&alert.hash())
+                {
+                    self.rmc.start_rmc(hash);
+                    if let Some(notification) = maybe_notification {
+                        self.send_notification_for_units(notification);
+                    }
+                }
+            }
+            AlertData::MultisignedHash(multisigned) => {
+                if let Some(notification) = self
+                    .multisigned_notifications
+                    .remove(multisigned.as_signable())
+                {
+                    self.send_notification_for_units(notification);
+                }
+            }
+        }
+    }
+
     pub async fn run(&mut self, mut handler: Handler<H, D, MK>, mut terminator: Terminator) {
         loop {
             futures::select! {
                 message = self.messages_from_network.next() => match message {
                     Some(message) => self.handle_message_from_network(&mut handler, message),
                     None => {
-                        error!(target: LOG_TARGET, "{:?} Message stream closed.", self.node_index);
+                        error!(target: LOG_TARGET, "{:?} message stream closed.", self.node_index);
```

Comment:
Hm, this reminds me, the fact that we log our own node index here is _extremely_ silly, since you are touching this maybe remove it?

Reply:
Ah, but now the logs should start with a capital letter (and generally be structured like sentences.

Reply:
aight